lib/rack/test.rb

```diff
@@ -217,31 +217,31 @@ def env_for(uri, env)
         # Stringifying and upcasing methods has be commit upstream
         env['REQUEST_METHOD'] ||= env[:method] ? env[:method].to_s.upcase : 'GET'
 
-        if %w[GET DELETE].include?(env['REQUEST_METHOD'])
+        params = env.delete(:params) { {} }
+
+        if env['REQUEST_METHOD'] == 'GET'
           # merge :params with the query string
-          if params = env[:params]
+          if params = params
```

Comment:
Would an `if params` do the same thing as this?

Reply:
Yup, just copy-paste error. Fixed!